### PR TITLE
ISPN-7030 Java Hot Rod client intelligence configuration

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-7030

### CODE REVIEW

**Comment by tristantarrant** [comment]
CI failures in the Kerberos client tests are related to the changes. Fixing...

**Comment by tristantarrant** [comment]
The KrbAuth failures have now been resolved. The remaining failures are unrelated.

**Comment by slaskawi** [comment]
Looking

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationChildBuilder.java [hunk 1]
```diff
@@ -56,11 +56,13 @@
     */
    ConfigurationBuilder balancingStrategy(Class<? extends FailoverRequestBalancingStrategy> balancingStrategy);
 
+   @Deprecated
```
Comment (slaskawi) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationChildBuilder.java]: Could you please add information why it's deprecated and what to use instead?

Comment (tristantarrant) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationChildBuilder.java]: I have done so. Hopefully there isn't too much detail :)

Comment (slaskawi) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationChildBuilder.java]: Awesome! thanks!

**Comment by slaskawi** [comment]
Looking

**Comment by slaskawi** [comment]
@tristantarrant It looks like there's a build failure:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project infinispan-client-hotrod: Compilation failure: Compilation failure:
[ERROR] /home/slaskawi/work/infinispan/infinispan/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransportFactory.java:[137,37] cannot find symbol
[ERROR] symbol:   method classLoader()
[ERROR] location: variable configuration of type org.infinispan.client.hotrod.configuration.Configuration
[ERROR] /home/slaskawi/work/infinispan/infinispan/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java:[382,65] cannot find symbol
[ERROR] symbol:   method classLoader()
[ERROR] location: variable template of type org.infinispan.client.hotrod.configuration.Configuration
[ERROR] -> [Help 1]

```

**Comment by tristantarrant** [comment]
Silly me, I committed a typo just now.

**Comment by tristantarrant** [comment]
It's fixed

**Comment by slaskawi** [comment]
Then I look at the CI results for this PR and master branch I noted those two are new:
```
EmbeddedCompatClientListenerWithDslFilterTest.testEventFilterChangingParameter 	
 NonIndexedRemoteContinuousQueryTest.testContinuousQuery 
```
http://ci.infinispan.org/viewLog.html?buildId=43255&buildTypeId=bt9

Are they related?

**Comment by gustavonalle** [comment]
No. https://github.com/infinispan/infinispan/pull/4601

**Comment by slaskawi** [comment]
Ok than... merging.

**Comment by slaskawi** [comment]
Thanks @tristantarrant

**Pull request merged**